### PR TITLE
Doc: decision tree regressor tutorial notebook

### PR DESCRIPTION
**Pull request by thomas-quadratic**
This PR introduces an example notebook of the usage of DecisionTreeRegressor in concrete-ml and its comparaison to scikit-learn.

Currently, there was no such example for DecisionTreeRegressor. We propose a tutorial similar to the one on DecisionTreeClassifier: 
https://github.com/zama-ai/concrete-ml/blob/release/0.6.x/docs/advanced_examples/DecisionTreeClassifier.ipynb

### CODE REVIEW

**Review by jfrery — COMMENTED** [review]
Thank you for the PR. Nice work! Before we go forward and open the PR on concrete-ml, we have a few proposition that could be taken into account:

- you seem to only use 6 bits for the model. But it could be the case that a lower or higher bit-width offer better accuracy. Maybe you could simply add this as a hyper-parameter in the grid search?
-  along with the previous point, it might be a good opportunity to explain what the n_bits actually do to the data i.e. quantization. 

Note: the FHE execution time is wrong as the overall time is divided by the full test set instead of the FHE test subset.
 
Thanks a again for your contribution!

**Comment by thomas-quadratic** [comment]
Thank you for your comments. I made a series of modifications to address the issues you raised:
- Moved to concrete-ml version 1.0
- Analysis includes a comparaison of n_bits = 6, 7 and 8
- Added a little discussion of quantization, and link to the documentation
- fixed the prediction time estimates

As it is, the stats are different now in version 1.0. What was observed is better prediction accuracy, but much slower compute-times.

**Comment by fd0r** [comment]
Hello @thomas-quadratic, thanks a lot for your contribution!

Could you now open a PR with the notebook in the folder docs/advanced_examples on the branch https://github.com/zama-ai/concrete-ml/tree/tutorials please?
Using [CapitalizedWords](https://en.wikipedia.org/wiki/Camel_case) for the name of the notebook.

Once the PR is merged in the branch with the other bounty contributions we will procede to merge it on main and your work will be featured in the next release!

**Comment by thomas-quadratic** [comment]
Great, I have created the PR on tutorials branch of concrete-ml repo !
